Plugins/GitUIPluginInterfaces/IGitModule.cs

```diff
@@ -29,6 +29,14 @@ public interface IGitModule
         /// <param name="remoteName">The current remote name.</param>
         /// <param name="newName">The new remote name.</param>
         string RenameRemote(string remoteName, string newName);
+
+        /// <summary>
+        /// Parses the revisionExpression as a git reference and returns an <see cref="ObjectId"/>./>
+        /// </summary>
+        /// <param name="revisionExpression">An expression like HEAD or commit hash that can be parsed as a git reference.</param>
```

Comment:
Git names this as commit-ish. https://git-scm.com/docs/gitglossary#gitglossary-aiddefcommit-ishacommit-ishalsocommittish

I suggest the following:
A Git commit (commit-ish: object name hash or expression like HEAD)